src/Microsoft.Azure.WebJobs.Host/Diagnostics/ExceptionFormatter.cs

```diff
@@ -26,92 +26,106 @@ public static class ExceptionFormatter
+                GetFormattedException(sb, aggregate.InnerExceptions[i]);
+                sb.AppendLine("<---");
             }
-
-            return text;
         }
 
-        private static string GetStackForException(Exception exception, bool includeMessageOnly)
+        private static void GetStackForException(StringBuilder sb, Exception exception, bool includeMessageOnly)
         {
             var message = exception.Message;
             var className = exception.GetType().ToString();
-            var stackText = message.Length <= 0
+            sb.Append(message.Length <= 0
                 ? className
-                : className + " : " + message;
+                : className + " : " + message);
             var innerException = exception.InnerException;
             if (innerException != null)
             {
                 if (includeMessageOnly)
                 {
                     do
                     {
-                        stackText += " ---> " + innerException.Message;
+                        sb.Append(" ---> ");
+                        sb.Append(innerException.Message);
                         innerException = innerException.InnerException;
                     }
                     while (innerException != null);
                 }
                 else
                 {
-                    stackText += $" ---> {GetFormattedException(innerException)} {Environment.NewLine}   End of inner exception";
+                    sb.Append(" ---> ");
+                    GetFormattedException(sb, innerException);
+                    sb.AppendLine();
+                    sb.Append("   End of inner exception");
                 }
             }
 
-            string stackTrace = GetAsyncStackTrace(exception);
-            if (!string.IsNullOrEmpty(stackTrace))
-            {
-                stackText += Environment.NewLine + stackTrace;
-            }
-
-            return stackText;
+            AddAsyncStackTrace(sb, exception);
         }
 
-        private static string GetAsyncStackTrace(Exception exception)
+        private static void AddAsyncStackTrace(StringBuilder sb, Exception exception)
         {
             var stackTrace = new StackTrace(exception, fNeedFileInfo: true);
             var stackFrames = stackTrace.GetFrames();
             if (stackFrames == null)
             {
-                return string.Empty;
+                return;
             }
 
-            var stackFrameLines = from frame in stackFrames
+            var toProcess = from frame in stackFrames
                                   let method = frame.GetMethod()
                                   let declaringType = method?.DeclaringType
                                   where ShouldShowFrame(declaringType)
-                                  select FormatFrame(method, declaringType, frame);
+                                  select (method: method, declaringType: declaringType, frame: frame);
```

Comment:
What if you kept FormatFrame here instead of creating the tuples and then iterating through them?

Reply:
Oh my 😅  I simplified it by pushing it down to `FormatFrame`